fast64_internal/oot/tools/quick_import.py

```diff
@@ -0,0 +1,115 @@
+import os
+import re
+
+import bpy
+
+from ..f3d.properties import OOTDLImportSettings
+from ..skeleton.properties import OOTSkeletonImportSettings
+from ..animation.properties import OOTAnimImportSettingsProperty
+
+
+class QuickImportAborted(Exception):
+    def __init__(self, message):
+        super().__init__(message)
+        self.message = message
+
+
+def quick_import_exec(context: bpy.types.Context, sym_name: str):
+    sym_name = sym_name.strip()
+    if sym_name == "":
+        raise QuickImportAborted("No symbol name given")
+    if not all(
+        (
+            "a" <= c <= "z"
+            or "A" <= c <= "Z"
+            or "0" <= c <= "9"
+            or c
+            in {
+                "_",
+            }
+        )
+        for c in sym_name
+    ):
+        raise QuickImportAborted("Symbol names only have characters a-zA-Z0-9_")
+
+    sym_def_pattern = re.compile(rf"([^\s]+)\s+{sym_name}\s*(\[[^\]]*\])?\s*=")
+
+    base_dir_p = Path(context.scene.ootDecompPath)
+    assets_objects_dir_p = base_dir_p / "assets" / "objects"
+
+    all_found_defs: dict[Path, list[tuple[str, str]]] = dict()
+
+    for dirpath, dirnames, filenames in os.walk(assets_objects_dir_p):
+        dirpath_p = Path(dirpath)
+        for filename in filenames:
+            file_p = dirpath_p / filename
+            # Only look into C files
+            if file_p.suffix != ".c":
+                continue
+            source = file_p.read_text()
+            # Simple check to see if we should look into this file any further
+            if sym_name not in source:
+                continue
+            found_defs = sym_def_pattern.findall(source)
+            print(file_p, f"{found_defs=}")
+            all_found_defs[file_p] = found_defs
+
+    # Ideally if for example sym_name was gLinkAdultHookshotTipDL,
+    # all_found_defs now contains:
+    # {Path('.../assets/objects/object_link_boy/object_link_boy.c'): [('Gfx', '[]')]}
+    # or with gButterflySkel:
+    # {Path('.../assets/objects/gameplay_field_keep/gameplay_field_keep.c'): [('SkeletonHeader', '')]}
+
+    if len(all_found_defs) == 0:
+        raise QuickImportAborted(f"Couldn't find a definition of {sym_name}")
+    if len(all_found_defs) > 1:
+        raise QuickImportAborted(
+            f"Found definitions of {sym_name} in several files: "
+            + ", ".join(str(p.relative_to(assets_objects_dir_p)) for p in all_found_defs.keys())
+        )
+    assert len(all_found_defs) == 1
+    sym_file_p, sym_defs = list(all_found_defs.items())[0]
+    if len(sym_defs) > 1:
+        raise QuickImportAborted(
+            f"Found several definitions of {sym_name} in {sym_file_p.relative_to(assets_objects_dir_p)}"
+        )
+
+    # We found a single definition of the symbol
+    sym_def_type, sym_def_array_decl = sym_defs[0]
+    is_array = sym_def_array_decl != ""
+    object_name = sym_file_p.relative_to(assets_objects_dir_p).parts[0]
+
+    if sym_def_type == "Gfx" and is_array:
```

Comment:
if I want to implement more quick imports in the future I just need to add an if statement in this block? I just wanna avoid dealing with regex if I can lol

Reply:
(I'm talking about the ifs from line 83)

Reply:
Yeah the regex above matches any type and this if chain is what actually matches the type to know how to import it
You can try importing a texture or collision header for example, there will be an error message with the type of the symbol